src/go/metadata/metadata_fetcher_test.go

```diff
@@ -309,22 +312,28 @@ func TestFetchGCPAttributes(t *testing.T) {
 
 		mf := NewMockMetadataFetcher(mockBaseUrl, time.Now())
 
-		attrs, err := mf.FetchGCPAttributes()
-		if err != nil && tc.expectedGCPAttributes != nil {
-			t.Errorf(errorTmpl, tc.desc, nil, err)
+		attrs, actualError := mf.FetchGCPAttributes()
+		// if expected error is set, actual error should be equal to the expected error
```

Comment:
* after second thought, please move this into a separate test like `TestFetchGCPAttributes_Errors` go/gotip/episodes/50
* make the retry setting part of testCase for better readability

Reply:
> * after second thought, please move this into a separate test like `TestFetchGCPAttributes_Errors` go/gotip/episodes/50

Added individual test case `TestFetchGCPAttributes_NoMetadataServer_FailsAllRetries`.
> * make the retry setting part of testCase for better readability

Retry setting is added in `metadata_fetcher_test_util.go`, which will be shared by all test cases.

Reply:
Prefer to explicitly set it in test setup so that we know what we are testing against